hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/web/SimpleHttpProxyHandler.java

```diff
@@ -95,6 +99,27 @@ public void exceptionCaught(ChannelHandlerContext ctx, Throwable cause) {
     }
   }
 
+  private static class SslRedirectRewriter extends ChannelInboundHandlerAdapter {
```

Comment:
javadocs, less about what happens but 
* why this is needed at all -and include the jira ID. 
* how this fits into the process. Specifically, it's processing the response so must happen in the response phase.

+ make final.